internal/exec/util/unit.go

```diff
@@ -153,24 +153,30 @@ func (ut Util) EnableUnit(enabledUnit string) error {
 }
 
 func (ut Util) DisableUnit(disabledUnit string) error {
-	// We need to delete any enablement symlinks for a unit before sending it to a
-	// preset directive. This will help to disable that unit completely.
-	// For more information: https://github.com/coreos/fedora-coreos-tracker/issues/392
-	// This is a short-term solution until the upstream systemd PR
-	// (https://github.com/systemd/systemd/pull/15205) gets accepted.
-	if err := ut.Logger.LogOp(
-		func() error {
-			args := []string{"--root", ut.DestDir, "disable", disabledUnit}
-			if output, err := exec.Command(distro.SystemctlCmd(), args...).CombinedOutput(); err != nil {
-				return fmt.Errorf("cannot remove symlink(s) for %s: %v: %q", disabledUnit, err, string(output))
-			}
-			return nil
-		},
-		"removing enablement symlink(s) for %q", disabledUnit,
-	); err != nil {
-		return err
+	args := []string{"--root", ut.DestDir, "is-enabled", disabledUnit}
+	// check if a unit is currently enabled to see if we need to disable it
```

Comment:
We were passed a specific unit, so:

```suggestion
	// check if the unit is currently enabled to see if we need to disable it
```